src/components/spatial/Spatial.js

```diff
@@ -4,16 +4,89 @@ import {
   ScatterplotLayer, PolygonLayer, COORDINATE_SYSTEM, BitmapLayer, BaseTileLayer,
 } from 'deck.gl';
 import { load } from '@loaders.gl/core';
-import { SelectablePolygonLayer } from '../../layers';
+import { Texture2D } from '@luma.gl/webgl';
+import GL from '@luma.gl/constants';
+import { slice, openArray } from 'zarr';
+import { SelectablePolygonLayer, XRLayer } from '../../layers';
 import { tileToBoundingBox, getTileIndices } from './tiling-utils';
 import { cellLayerDefaultProps, PALETTE, DEFAULT_COLOR } from '../utils';
 import AbstractSelectableComponent from '../AbstractSelectableComponent';
 import LayersMenu from './LayersMenu';
 
+const zarrArrays = {};
+
 export function square(x, y, r) {
   return [[x, y + r], [x + r, y], [x, y - r], [x - r, y]];
 }
 
+async function getTexture(config, gl, tileSize, x, y, z, stride, width) {
+  const arrSlice = slice(stride * width * y + stride * x, stride * width * y + stride * (x + 1));
+  const zarrKey = config.zarrConfig.store + config.zarrConfig.path;
+  if (!zarrArrays[zarrKey]) {
+    zarrArrays[zarrKey] = await openArray(config.zarrConfig);
+  }
+  const arr = zarrArrays[zarrKey];
+  const dataSlice = await arr.get([arrSlice]);
+  const { data } = dataSlice;
+  // eslint-disable-next-line no-nested-ternary
+  const formats = data instanceof Uint8Array
+    ? {
+      format: GL.R8UI,
+      dataFormat: GL.RED_INTEGER,
+      type: GL.UNSIGNED_BYTE,
+    }
+    : (
+      data instanceof Uint16Array
+        ? {
+          format: GL.R16UI,
+          dataFormat: GL.RED_INTEGER,
+          type: GL.UNSIGNED_SHORT,
+        }
+        : {
+          format: GL.R32UI,
+          dataFormat: GL.RED_INTEGER,
+          type: GL.UNSIGNED_INT,
+        }
+    );
+  const { channelType } = config;
+  const texObj = {};
+  texObj[channelType] = new Texture2D(gl, {
+    width: tileSize,
+    height: tileSize,
+    data,
+    // we don't want or need mimaps
+    mipmaps: false,
+    parameters: {
+      // NEAREST for integer data
+      [GL.TEXTURE_MIN_FILTER]: GL.NEAREST,
+      [GL.TEXTURE_MAG_FILTER]: GL.NEAREST,
+    },
+    ...formats,
+  });
+  return texObj;
+}
+
+function loadZarr(sourceChannels, tileSize, x, y, z, width, gl) {
+  const zoom = z;
+  const textureNames = ['redTexture', 'greenTexture', 'blueTexture'];
+  const configList = sourceChannels.map((channel, i) => ({
```

Comment:
So for now, limited to three channels in RGB order? When there are more, will the change be here, or somewhere else?

Reply:
Yes, this is only temporary.